client/my-sites/plan-features/index.jsx

```diff
@@ -481,6 +563,53 @@ export class PlanFeatures extends Component {
 		} );
 	}
 
+	handleUpgradeClick( singlePlanProperties ) {
+		const {
+			displayJetpackPlans,
+			isInSignup,
+			selectedSiteSlug,
+			withDiscount,
+			onUpgradeClick: ownPropsOnUpgradeClick,
+		} = this.props;
+
+		const {
+			availableForPurchase,
+			cartItemForPlan,
+			isPrivateAndGoingAtomic,
+			planPath,
+			productSlug,
+		} = singlePlanProperties;
+
+		if ( ownPropsOnUpgradeClick && ownPropsOnUpgradeClick !== noop && cartItemForPlan ) {
+			ownPropsOnUpgradeClick( cartItemForPlan );
+			return;
+		}
+
+		if ( ! availableForPurchase ) {
+			return;
+		}
+
+		const args = {};
+		// Auto-apply the coupon code to the cart for WPCOM sites
+		if ( ! displayJetpackPlans && withDiscount ) {
+			args.coupon = withDiscount;
+		}
+
+		if ( isPrivateAndGoingAtomic ) {
```

Comment:
What do you think to moving this conditional up above `const args = ...` to keep the args logic together with its usage in `page( addQuery`?